File: core-server/backend_server/backend_server/Models/DomainModels/Train.cs
﻿namespace backend_server.Models.DomainModels;

public sealed class Train : BaseEntity
{
    public string Name { get; set; }

    public string Model { get; set; }

    public string DriverName { get; set; }

    public int Contact { get; set; }

    public int NoOfSeats { get; set; }

    public string StartingStation { get; set; }

    public string EndingStation { get; set; }

    public List<TrainSchedule> Schedules { get; set; }
}

File: mobile/app/src/main/java/com/example/mobile_app/models/Train.java
package com.example.mobile_app.models;

public class Train {
    public String id, trainName, model, driverName, contactNumber, startingStation, endingStation;
    public int numberOfSeats;
    public Train(String id, String trainName, String model, String driverName, String contactNumber, String startingStation, String endingStation, int numberOfSeats) {
        this.id = id;
        this.trainName = trainName;
        this.model = model;
        this.driverName = driverName;
        this.contactNumber = contactNumber;
        this.startingStation = startingStation;
        this.endingStation = endingStation;
        this.numberOfSeats = numberOfSeats;
    }

}
